typescript/sdk/src/providers/MultiProvider.ts

```diff
@@ -39,12 +39,23 @@ export class MultiProvider {
     chainMetadata: ChainMap<ChainMetadata> = defaultChainMetadata,
     options: MultiProviderOptions = {},
   ) {
-    this.metadata = chainMetadata;
+    for (const chain of Object.values(chainMetadata)) {
+      this.addChain(chain);
+    }
+    this.logger = debug(options?.loggerName || 'hyperlane:MultiProvider');
+  }
+
+  addChain(metadata: ChainMetadata): void {
```

Comment:
You'll need to check if useSharedSigner is true and if so add an entry in `this.signers` for the new chain. I realize the whole shared signer thing is awkward but it was a requirement in some places.